src/test/clojure/xtdb/sql_test.clj

```diff
@@ -2514,3 +2514,17 @@ UNION ALL
 
     (t/is (= {:app_count 2}
              (jdbc/execute-one! tu/*node* ["SELECT COUNT(*) app_count FROM applications"])))))
+
+(t/deftest test-order-nulls-first-last
+  (xt/execute-tx tu/*node* ["INSERT INTO foo (_id, name) VALUES (1, 'foo'), (2, 'bar'), (3, 'baz')"])
+  (xt/execute-tx tu/*node* ["INSERT INTO foo (_id, name) VALUES (1, 'fooz'), (2, 'barz'), (3, 'bazz')"])
+  (t/is (=
+         [nil nil nil]
```

Comment:
incorrect code (e.g. returning all nulls, or not sorting non-nulls times correctly) would also pass this test - could you explicitly specify enough to prove this is behaving correctly?

it should be the same valid-to every time, if that's what you were concerned about - I think this namespace has a mock-clock fixture